tests/utilities/test_auto_restart.py

```diff
@@ -1195,6 +1196,29 @@ def test_auto_restart_under_signal(on_last_batch, val_check_interval, failure_on
         assert "dataloader_state_dict" in state_dict
 
 
+def test_supports_state_dict_protocol():
```

Comment:
If I understand correctly, it is also going to fail for incompatible signature, correct?

Reply:
isinstance would return False with incompatible signature.